docs/cli-tool/kustomize-plugin.md

```diff
@@ -0,0 +1,402 @@
+kind: ResourceDistributionGenerator
+metadata:
+  ... ... 
+resource:
+  ... ...
+  resourceOptions:
+    annotations:
+      dashboard: "1"
+    labels:
+      environment: "dev"
+targets:
+	... ...
+options:
+  annotations:
+    type: "slave"
+  labels:
+    version: "stable"
+```
+
+### A Complete Use Case
+
+1. Create a demo directory as a workspace and enter. Place the downloaded ResourceDistributionGenerator plugin in the current directory and enter the following command to create a configuration file named rdGenerator.yaml.
+
+```yaml
+cat > rdGenerator.yaml <<EOF
+#rdGenerator.yaml
+apiVersion: apps.kruise.io/v1alpha1
+kind: ResourceDistributionGenerator
+metadata:
+  name: rdname
+  annotations:
+    config.kubernetes.io/function: |
+      exec:
+        path: ./resourcedistributiongenerator
+resource:
+  resourceKind: ConfigMap
+  resourceName: cmname
+  files:
+    - application.properties
+  literals:
+    - JAVA_HOME=/opt/java/jdk
+  resourceOptions:
+    annotations:
+      dashboard: "1"
+options:
+  labels:
+    app.kubernetes.io/name: "app1"
+targets:
+  includedNamespaces:
+    - ns-1
+  namespaceLabelSelector:
+    matchLabels:
+      group: "test"
+EOF
+```
+
+2. Create the application.properties file using the following command as file input.
+
+```properties
+cat > application.properties <<EOF
+FOO=Bar
+FIRST=1
+SECOND=2
+LAST=3
+EOF
+```
+
+3. Create the kustomization file with the following command.
+
+```yaml
+cat > kustomization.yaml <<EOF
+#kustomization.yaml
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+generators:
+- rdGenerator.yaml
+EOF
+```
+
+4. Use the `kustomize build --enable-alpha-plugins --enable-exec .` command to build application, the effect is as follows
+
+```yaml
+apiVersion: apps.kruise.io/v1alpha1
+kind: ResourceDistribution
+metadata:
+  labels:
+    app.kubernetes.io/name: app1
+  name: rdname
+spec:
+  resource:
+    apiVersion: v1
+    data:
+      JAVA_HOME: /opt/java/jdk
+      application.properties: |
+        FOO=Bar
+        FIRST=1
+        SECOND=2
+        LAST=3
+    kind: ConfigMap
+    metadata:
+      annotations:
+        dashboard: "1"
+      name: cmname
+  targets:
+    includedNamespaces:
+      list:
+      - name: ns-1
+    namespaceLabelSelector:
+      matchLabels:
+        group: test
+```
+
+### Use the ResourceDistribution Generator in ArgoCD
+
+In argocd, the usage of the kustomize plugin is the same as above. In addition to that, add a build option to kustomize that allows third-party plugins. Find the configMap named argocd-cm in the kubernetes cluster and add the following to the `data` field `kustomize.buildOptions : --enable-alpha-plugins --enable-exec` to add build options for third-party plugins to the default version of kustomize. See [ArgoCD](https://argo-cd.readthedocs.io/en/stable/user-guide/kustomize/#kustomize-build-optionsparameters) for more information.
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+    ....
```

Comment:
apiVersion: v1
kind: ConfigMap
metadata:
  name: argocd-cm
  namespace: argocd
  labels:
    app.kubernetes.io/name: argocd-cm
    app.kubernetes.io/part-of: argocd
data:
    kustomize.buildOptions: --enable-alpha-plugins --enable-exec